Setup Authentication and Dashboard Page

ogwurujohnson opened the pull request:
# Description

This PR, creates the Dashboard component and also tries to implement and consume the signup endpoint by creating reducers and action types

Fixes # (issue)



## Type of change


Please add a relevant label.


## Change status
 - Complete but not tested

# How Has This Been Tested?
 - By running on the browser

Please describe steps taken to ensure this feature is functional and does not break other features:
 - Before making any push I pull from my colleague's branch 

# Checklist:



- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation, if relevant

- [x] I have run the app using my feature and ensured that no functionality is broken

- [x] My changes generate no new warnings

- [x] There are no merge conflicts

annawinther committed:
installed deps, initial commit

annawinther committed:
added new container component

annawinther committed:
commented out title in App.js

annawinther committed:
building the login component

annawinther committed:
added BrowserRouter and Route for the login component

annawinther committed:
added a sign up form on the same page as the login page

ogwurujohnson committed:
Feat: Introduce routing to application

ogwurujohnson committed:
Feat: Flesh out the Add event component

ogwurujohnson committed:
Feat: flesh out the Evevnts component

ogwurujohnson committed:
Feat: Flesh out the Dashboard container and rendering the individual components relying on it

annawinther committed:
made some changes to the login component

ogwurujohnson committed:
Feat: Create auth action types

annawinther committed:
sign-up and log-in are in two separate components

ogwurujohnson committed:
Feat: Build out the auth reducer taking care of each available case change dependent on the action type

ogwurujohnson committed:
Feat: Use combine reducers on the root of the reducers folder

ogwurujohnson committed:
Feat: Create store for our application, and setup the redux devtools

ogwurujohnson committed:
Feat: Create and use middleware for setting token returned  during a successful login or signin to local storage

ogwurujohnson committed:
Feat: Create a util file for localstorage operations to promote reusability

annawinther committed:
adding axios requet to log in and register

annawinther committed:
added axios requests to the login and signup

annawinther committed:
still working on the login an dsignup action creators

annawinther committed:
error corrected

annawinther committed:
connected Login component to the action creator, trying to get it working correctly

ogwurujohnson committed:
Feat: Create the handleChange method to set to state the individual values coming from the input field

ogwurujohnson committed:
Feat: Create the handleSubmit method for calling our action creators and passing to it an object of credentials

ogwurujohnson committed:
Feat: connect the component to the application store

annawinther committed:
refactored the action creator, not fully working yet

annawinther committed:
bug fixed, now working

annawinther committed:
added return statement in signup action creator

DianaIgnatescu commented on src/components/AddEvent.js:
```diff
@@ -0,0 +1,16 @@
+    render() {
+        return (
+            <div>
+                <p>Add new event</p>
```
Try and have a look at the Seeding to get a sense of what inputs are expected

ogwurujohnson replied:
fixed

DianaIgnatescu commented on src/components/Login.js:
```diff
@@ -0,0 +1,26 @@
+            <LoginWrapperStyled> 
+                <h3>Log in here</h3>
+                <div>
+                <input type="text" name="name" placeholder="Name" required />
```
Pay attention to indentation as this can make it harder to read.

ogwurujohnson replied:
fixed

DianaIgnatescu commented on src/components/Login.js:
```diff
@@ -0,0 +1,26 @@
+                <input type="text" name="name" placeholder="Name" required />
+                </div>
+                <div>
+                <input type="passowrd" name="password" placeholder="Password" required />
```
typo here

DianaIgnatescu commented on src/containers/Dashboard.js:
```diff
@@ -0,0 +1,14 @@
+            </div>
+        )
+    }
+}
```
I think this should be moved inside the Views directory. Same goes for the Login and Register.

ogwurujohnson replied:
fixed

DianaIgnatescu reviewed (changes requested):
This looks good overall. I have made some small comments that will hopefully help us stay organized since the current structure is a bit difficult to follow.  
Can we please move the actions and reducers directories inside a store directory?
Also the Login and Dashboard should be moved inside the views directory.

ogwurujohnson committed:
Feat: Redirect user to login page after successful signup

annawinther committed:
getting token saved to Local Storage now, all works

annawinther committed:
changes made as requested. Should be working fine now

ogwurujohnson committed:
Fix: Rearrange folder structure to fit team lead recommendation
 - move login, dashboard and signup pages to view directory
 - move action and reducers directory to store directory and fix typos

ogwurujohnson committed:
Fix: Make change to the input field for adding a new event

ogwurujohnson committed:
Fix: refactor check to remove error

ogwurujohnson committed:
Fix: pass in the token as payload

ogwurujohnson committed:
Feat: dispatch action when there is an error making an axios call

ogwurujohnson committed:
Feat: Redirect user to dashboard page after successful login

DianaIgnatescu reviewed (approved):
Looks good to me!
